tests/integrationv2/test_record_padding.py

```diff
@@ -0,0 +1,161 @@
+import copy
+import pytest
+import re
+
+from configuration import available_ports, TLS13_CIPHERS, ALL_TEST_CURVES, MINIMAL_TEST_CERTS
+from common import ProviderOptions, Protocols, data_bytes
+from fixtures import managed_process  # lgtm [py/unused-import]
+from providers import Provider, S2N, OpenSSL
+from utils import invalid_test_parameters, get_parameter_name, get_expected_s2n_version, to_bytes
+
+PADDING_SIZE_SMALL = 250
+PADDING_SIZE_MEDIUM = 1000
+PADDING_SIZE_MAX = 1 << 14
+
+PADDING_SIZES = [
+    PADDING_SIZE_SMALL,
+    PADDING_SIZE_MEDIUM,
+    PADDING_SIZE_MAX
+]
+
+# arbitrarily large payload size
+PAYLOAD_SIZE = 1024
+
+# regex that matches server/client 'write to' for application records (with tls header).
+# Group 0: The 'write to ...' line, denoting openssl is sending a record
+# Group 1: The start of an application header denoted by type 0x17
+# Group 2: The size of the record expressed in two bytes
+RECORD_SIZE_GROUP = 2
+OPENSSL_WRITE_TO_REGEX = r"(write to [^\\n]*)(\\n0000 - 17 03 03 )([0-9a-f]{2} [0-9a-f]{2})"
+
+
+def get_payload_size_from_openssl_trace(record: str) -> int:
```

Comment:
Nit: "record" is a deceptive name for this variable. Maybe record_size_hex? size_bytes?